generic-sensor/generic-sensor-tests.js

```diff
@@ -146,49 +142,44 @@ function runGenericSensorTests(sensorType) {
     }
   }, `${sensorType.name}: no exception is thrown when calling stop() on already stopped sensor`);
 
-  promise_test(() => {
-    return new Promise((resolve,reject) => {
-      let iframe = document.createElement('iframe');
-      iframe.srcdoc = '<script>' +
-                      '  window.onmessage = message => {' +
-                      '    if (message.data === "LOADED") {' +
-                      '      try {' +
-                      '        new ' + sensorType.name + '();' +
-                      '        parent.postMessage("FAIL", "*");' +
-                      '      } catch (e) {' +
-                      '        parent.postMessage(e.name, "*");' +
-                      '      }' +
-                      '    }' +
-                      '   };' +
-                      '<\/script>';
-      iframe.onload = () => iframe.contentWindow.postMessage('LOADED', '*');
-      document.body.appendChild(iframe);
-      window.onmessage = message => {
-        if (message.data == 'SecurityError') {
-          resolve();
-        } else {
-          reject();
-        }
-      }
-    });
+  promise_test(async t => {
+    const iframe = document.createElement('iframe');
+    iframe.srcdoc = '<script>' +
+                    '  window.onmessage = message => {' +
+                    '    if (message.data === "LOADED") {' +
+                    '      try {' +
+                    '        new ' + sensorType.name + '();' +
+                    '        parent.postMessage("FAIL", "*");' +
+                    '      } catch (e) {' +
+                    '        parent.postMessage(e.name, "*");' +
+                    '      }' +
+                    '    }' +
+                    '   };' +
+                    '<\/script>';
+    iframe.onload = () => iframe.contentWindow.postMessage('LOADED', '*');
+    document.body.appendChild(iframe);
+    const eventWatcher = new EventWatcher(t, window, "message");
+    const message = await eventWatcher.wait_for("message");
+    assert_equals(message.data, 'SecurityError');
   }, `${sensorType.name}: throw a 'SecurityError' when constructing sensor object within iframe`);
 
-  async_test(t => {
-    let sensor = new sensorType();
-    sensor.onreading = t.step_func(() => {
-      assert_reading_not_null(sensor);
-      let cachedSensor1 = reading_to_array(sensor);
-      let win = window.open('', '_blank');
-      t.step_timeout(() => {
-        let cachedSensor2 = reading_to_array(sensor);
-        win.close();
-        sensor.stop();
-        assert_array_equals(cachedSensor1, cachedSensor2);
-        t.done();
-      }, 1000);
-    });
-    sensor.onerror = t.step_func_done(unreached);
+  promise_test(async t => {
+    const sensor = new sensorType();
+    const eventWatcher = new EventWatcher(t, sensor, ["reading", "error"]);
```

Comment:
How about use `sensorWatcher` to distinguish from `visibilityChangeWatcher` ?

Reply:
Modify eventWatcher to sensorWatcher, thanks.